core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -297,7 +297,7 @@ public TableDataMap registerDataMap(CarbonTable table,
         dataMapSchema, dataMapFactory, blockletDetailsFetcher, segmentPropertiesFetcher);
 
     tableIndices.add(dataMap);
-    allDataMaps.put(tableUniqueName, tableIndices);
+    allDataMaps.put(tableUniqueName.toLowerCase(), tableIndices);
```

Comment:
The name is already lowercased here, no need to convert it here. If it is not, there maybe other bugs that cause it.

Reply:
I found the actual bug, for external table, during table creation itself table name and db name was not converted to lowercase, because of this if table name is not in lowercase, when drop table, blocklet datamap was not getting cleared.